compiler/testData/diagnostics/tests/classLiteral/smartCast.fir.kt

```diff
@@ -17,7 +17,7 @@ fun test2(): KClass<out Foo>? {
     var f: Foo? = null
     if (f != null) {
         run { f = null }
-        return <!RETURN_TYPE_MISMATCH!><!EXPRESSION_OF_NULLABLE_TYPE_IN_CLASS_LITERAL_LHS!>f<!>::class<!>
+        return <!EXPRESSION_OF_NULLABLE_TYPE_IN_CLASS_LITERAL_LHS!>f<!>::class
```

Comment:
I don't understand how your checker could provoke this change (will recheck myself)